src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -113,3 +173,241 @@ def delete_model(self, model_name: str, model_version: str = "*") -> None:
+                        "Traceback": "".join(
+                            traceback.TracebackException.from_exception(exception).format()
+                        ),
+                    }
+                )
+        if failed_imports:
+            raise RuntimeError(
+                f"Failures when importing models to curated hub in parallel: {failed_imports}"
+            )
+
+    def _sync_public_model_to_hub(self, model: JumpStartModelInfo, thread_num: int):
+        """Syncs a public JumpStart model version to the Hub. Runs in parallel."""
+        model_specs = utils.verify_model_region_and_return_specs(
+            model_id=model.model_id,
+            version=model.version,
+            region=self.region,
+            scope=JumpStartScriptScope.INFERENCE,
+            sagemaker_session=self._sagemaker_session,
+        )
+        studio_specs = self._fetch_studio_specs(model_specs=model_specs)
+
+        dest_location = S3ObjectLocation(
+            bucket=self.hub_bucket_name, key=f"{model.model_id}/{model.version}"
+        )
+        src_files = file_generator.generate_file_infos_from_model_specs(
+            model_specs, studio_specs, self.region, self._s3_client
+        )
+        dest_files = file_generator.generate_file_infos_from_s3_location(
+            dest_location, self._s3_client
+        )
+
+        comparator = SizeAndLastUpdatedComparator()
+        sync_request = HubSyncRequestFactory(
+            src_files, dest_files, dest_location, comparator
+        ).create()
+
+        if len(sync_request.files) > 0:
+            MultiPartCopyHandler(
+                thread_num=thread_num,
+                sync_request=sync_request,
+                region=self.region,
+                label=dest_location.key,
+            ).execute()
+        else:
+            JUMPSTART_LOGGER.warning("Nothing to copy for %s v%s", model.model_id, model.version)
+
+        # TODO: Tag model if specs say it is deprecated or training/inference
+        # vulnerable. Update tag of HubContent ARN without version.
+        # Versioned ARNs are not onboarded to Tagris.
+        tags = []
+
+        search_keywords = [
+            f"{JUMPSTART_HUB_MODEL_ID_TAG_PREFIX}:{model.model_id}",
+            f"{JUMPSTART_HUB_MODEL_VERSION_TAG_PREFIX}:{model.version}",
+            f"{FRAMEWORK_TAG_PREFIX}:{model_specs.get_framework()}",
+            f"{TASK_TAG_PREFIX}:TODO: pull from specs",
+        ]
+
+        hub_content_document = str(HubContentDocument_v2(spec=model_specs))
+
+        self._sagemaker_session.import_hub_content(
+            document_schema_version=HubContentDocument_v2.SCHEMA_VERSION,
+            hub_content_name=model.model_id,
+            hub_content_version=model.version,
+            hub_name=self.hub_name,
+            hub_content_document=hub_content_document,
+            hub_content_type=HubContentType.MODEL,
+            hub_content_display_name="",
+            hub_content_description="",
```

Comment:
Use [this attribute](https://github.com/aws/sagemaker-model-hub/blob/5f4be7241f1a9bcb47d00c203ea0483bb56e433a/src/sagemakermodelhub/content_generation/metadata/generated_metadata/studio/manifests/studio_models_manifest.json#L13) please.